app/src/main/assets/native/MediaSegmentsPlugin.js

```diff
@@ -0,0 +1,32 @@
+export class MediaSegmentsPlugin {
+    SETTING_PREFIX = 'segmentTypeAction';
+
+    constructor({ events, appSettings, dashboard }) {
+        this.appSettings = appSettings;
+        this.dashboard = dashboard;
+
+        events.on(appSettings, 'change', (_, name) => this.onSettingsChanged(name));
```

Comment:
I think it would be better if we also read all settings when the webview launches, so probably here in the plugin constructor.
Otherwise, if you configure media segments now (while the app doesn't support it) they won't work like configured when the app is updated with these changes.